packages/boxplot/src/compute/legends.ts

```diff
@@ -0,0 +1,37 @@
+import {
+    BoxPlotDatum,
+    BoxPlotLegendProps,
+    BoxPlotSvgProps,
+    ComputedBoxPlotSummary,
+    LegendData,
+} from '../types'
+import { getPropertyAccessor } from '@nivo/core'
+import { uniqBy } from 'lodash'
+
+export const getLegendData = <RawDatum extends BoxPlotDatum>({
+    boxPlots,
+    dataFrom,
+    direction,
+    legendLabel,
+}: {
+    boxPlots: ComputedBoxPlotSummary[]
+    dataFrom: BoxPlotLegendProps['dataFrom']
+    direction: string
+    legendLabel: BoxPlotSvgProps<RawDatum>['legendLabel']
+}) => {
+    const legendFrom = dataFrom === 'subGroups' ? 'subGroup' : 'group'
+    const getLegendLabel = getPropertyAccessor(legendLabel ?? legendFrom)
+    const getFromId = typeof dataFrom === 'function' ? dataFrom : getPropertyAccessor(legendFrom)
+    return uniqBy(
+        boxPlots.map(
+            boxPlot =>
+                ({
+                    id: getFromId(boxPlot.data),
+                    color: boxPlot?.color,
+                    direction,
+                    label: getLegendLabel(boxPlot?.data),
```

Comment:
Sorry but I don't understand why we have `getFromId(boxPlot.data)` VS `getLegendLabel(boxPlot?.data)`, what am I missing 😅 ?

Reply:
I simplified this so that only getFromLabel is used. After changes, the id comes from groups or subGroups, which helps ensure that the legends appear in the order specified in the props.

Related to this, the legends spec now no longer uses 'dataFrom'.

(Coincidentally, the 'dataFrom' was borrowed from the Bar package. It is probably not needed there either.)

Reply:
Yea, I'm pretty sure I would have a lot of comments if I was to review my own code right now...